### ISSUE TITLE
Search by multiple Message-Id's

### ISSUE THREAD

**Issue by M1z23R** [mscdex/node-imap#903]
I am working on groups of sent emails, so I want to be able to find all replies to those sent messages. I know I can query by in-reply-to and references, I also know, I can chain multiple OR*s, but I am not sure what the limit would be, and I don't want to bombard the mail server with a lot of data just for the search query since - if I am not mistaking, those queries are executed on server and not client side - correct me if I am wrong.
Basically, what I am looking for is a way to pass an array of message-id's just like passing array of arrays of uid's.

**Comment by M1z23R** [comment]
I just tried with 500 emails, and to be honest, it doesn't seem that slow or anything. I'd still like to hear some other thoughts, but here is how I did it in case someone else needs it:

> 
>     var searchCriteria: any = [["HEADER", "IN-REPLY-TO", messageIds[0]]];
>     if (messageIds.length > 1) {
>       let indx = 1;
>       while (indx !== messageIds.length - 1) {
>         searchCriteria = [
>           [
>             "OR",
>             ["HEADER", "IN-REPLY-TO", messageIds[indx]],
>             ...searchCriteria,
>           ],
>         ];
>         indx++;
>       }
>     }
>
